apps/sms/views/shared/js/settings.js

```diff
@@ -72,6 +79,15 @@ var Settings = {
     }
   },
 
+  // return resolve promise with sendReadReport status
+  initSendReadReport: function initSendReadReport(value) {
+    this._sendReadReportDefer.resolve(value);
+    delete this._sendReadReportDefer;
+
+    navigator.mozSettings.addObserver('messages.mms.sendReadReport.enabled',
+      (e) => this.sendReadReport = Promise.resolve(e.settingValue));
```

Comment:
BTW I just wondering why don't we simply call the Settings get API directly but  use init here. I know it would be replaced once we introduce the settings client, but shouldn't it be easier to call the settings get API in sendReadReport?